youxia-deployer/src/main/java/io/cloudbindle/youxia/deployer/Deployer.java

```diff
@@ -192,25 +203,80 @@ public Deployer(String[] args) {
     /**
      * Determine the number of clients that we need to spawn.
      *
-     * @return
+     * @return return the flavour -> number of clients that we can start up this run
      */
-    private int assessClients() {
+    private ImmutablePair<String, Integer> assessClients() {
         AbstractInstanceListing lister;
+        String defaultFlavour;
         if (options.has(this.openStackModeSpec)) {
             lister = ListingFactory.createOpenStackListing();
+            defaultFlavour = youxiaConfig.getString(DEPLOYER_OPENSTACK_FLAVOR);
+
         } else if (options.has(this.azureModeSpec)) {
             lister = ListingFactory.createAzureListing();
+            defaultFlavour = youxiaConfig.getString(DEPLOYER_AZURE_FLAVOR);
         } else {
             lister = ListingFactory.createAWSListing();
+            defaultFlavour = youxiaConfig.getString(DEPLOYER_INSTANCE_TYPE);
         }
+        assert (defaultFlavour != null);
         Map<String, InstanceDescriptor> map = lister.getInstances();
         Log.info("Found " + map.size() + " clients");
+        final Integer batchSize = options.valueOf(this.batchSizeSpec);
+
+        LinkedHashMap<String, Integer> clientTypes = new LinkedHashMap<>();
+        if (options.has(totalNodesSpec)) {
+            int totalNumNodes = options.valueOf(totalNodesSpec);
+            int clientsNeeded = totalNumNodes - map.size();
+            Log.info("Need " + clientsNeeded + " more workers");
+            int clientsAfterBatching = Math.min(batchSize, clientsNeeded);
+            Log.info("After batch limit, we can requisition up to " + clientsAfterBatching + " this run");
+            clientTypes.put(defaultFlavour, clientsAfterBatching);
+        } else {
+            assert (options.has(instanceRestrictions));
```

Comment:
Should this be here, or should there be an if-statement that throws an exception for this condition, with a user-friendly message?